### ISSUE TITLE
Potential new features

### ISSUE THREAD

**Issue by asmahani** [asmahani/ordinal-boost#6]
- [x] plotting utility (e.g. training and holdout errors vs. iteration no)
- [x] cross-validation-based early stopping
- [x] line search (for both g (gamma) and theta)
- [x] handling of invalid thresholds (seems a rare issue)
- [x] rather than hard-coding pdf/cdf (currently normal distro, i.e., we have ordered probit), accept them as input

**Comment by asmahani** [comment]
Re line search, we implemented it for theta, but decided to skip it for gamma at the moment, which means we always use gamma=1.0. however, we will keep the learning rate / shrinkage parameter for the regression function, since otherwise it will rapidly overfit (i.e., when learning rate is fixed at zero).

**Comment by asmahani** [comment]
closing the link function issue as we turned it into a separate item/issue.
